packages/server/src/jsRunner/vm/isolated-vm.ts

```diff
@@ -97,10 +97,14 @@ export class IsolatedVM implements VM {
     return this
   }
 
-  withContext(context: Record<string, any>) {
+  withContext<T>(context: Record<string, any>, f: () => T) {
     this.addToContext(context)
 
-    return this
+    try {
+      return f()
+    } finally {
+      this.removeFromContext(Object.keys(context))
```

Comment:
Might this be deleting staff that was defined somewhere else?

Reply:
It's not deleting anything that wouldn't have already been overridden when we set the context, so I figured it's not a concern. Open to situations where it might be, though.